src/server/editorServices.ts

```diff
@@ -1974,7 +1974,7 @@ namespace ts.server {
             // Read updated contents from disk
             const configFilename = normalizePath(project.getConfigFilePath());
 
-            const configFileContent = this.host.readFile(configFilename)!; // TODO: GH#18217
+            const configFileContent = this.host.readFile(configFilename) || "";

```

Comment:
I usually think of `{}` as the simplest tsconfig.  What does `parseJsonText` return when it's empty?  I some sort of diagnostic produced?

Reply:
It will create source file with no statements and parseJsonSourceFileConfigFileContent will take that to mean `{}`

Reply:
Cool.  Will some sort of diagnostic be produced for the failed read?

Reply:
created #37611